src/ucp/rma/rma_send.c

```diff
@@ -252,14 +252,19 @@ ucs_status_ptr_t ucp_put_nbx(ucp_ep_h ep, const void *buffer, size_t count,
             goto out_unlock;
         }
 
-        datatype = ucp_request_param_datatype(param);
         req = ucp_request_get_param(worker, param,
                                     {ret = UCS_STATUS_PTR(UCS_ERR_NO_MEMORY);
                                     goto out_unlock;});
         req->send.rma.rkey        = rkey;
         req->send.rma.remote_addr = remote_addr;
-        if (UCP_DT_IS_CONTIG(datatype)) {
-            contig_length = ucp_contig_dt_length(datatype, count);
+
+        if (ucs_likely(param->op_attr_mask == 0)) {
```

Comment:
mask with (UCP_OP_ATTR_FIELD_DATATYPE|NO_IMM_CMPL) like in tag_send.c

Reply:
Fixed